File: generated-code/codecraft/rust/src/model/player.rs
use super::*;

/// Player (strategy, client)
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub struct Player {
    /// Player's ID
    pub id: i32,
    /// Current score
    pub score: i32,
    /// Current amount of resource
    pub resource: i32,
}

impl trans::Trans for Player {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        self.id.write_to(writer)?;
        self.score.write_to(writer)?;
        self.resource.write_to(writer)?;
        Ok(())
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let id: i32 = trans::Trans::read_from(reader)?;
        let score: i32 = trans::Trans::read_from(reader)?;
        let resource: i32 = trans::Trans::read_from(reader)?;
        Ok(Self {
            id,
            score,
            resource,
        })
    }
}

File: generated-code/codecraft/ruby/model/player.rb
# Player (strategy, client)
class Player
    # Player's ID
    attr_accessor :id
    # Current score
    attr_accessor :score
    # Current amount of resource
    attr_accessor :resource

    def initialize(id, score, resource)
        @id = id
        @score = score
        @resource = resource
    end

    # Read Player from input stream
    def self.read_from(stream)
        id = stream.read_int()
        score = stream.read_int()
        resource = stream.read_int()
        Player.new(id, score, resource)
    end

    # Write Player to output stream
    def write_to(stream)
        stream.write_int(@id)
        stream.write_int(@score)
        stream.write_int(@resource)
    end

    def to_s
        string_result = "Player { "
        string_result += "id: "
        string_result += @id.to_s
        string_result += ", "
        string_result += "score: "
        string_result += @score.to_s
        string_result += ", "
        string_result += "resource: "
        string_result += @resource.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end